server/odc-service/src/main/java/com/oceanbase/odc/service/sqlcheck/factory/SQLAffectedRowsFactory.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2023 OceanBase.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.oceanbase.odc.service.sqlcheck.factory;
+
+import java.util.Map;
+
+import org.springframework.jdbc.core.JdbcOperations;
+
+import com.oceanbase.odc.core.shared.constant.DialectType;
+import com.oceanbase.odc.service.sqlcheck.SqlCheckRule;
+import com.oceanbase.odc.service.sqlcheck.SqlCheckRuleFactory;
+import com.oceanbase.odc.service.sqlcheck.model.SqlCheckRuleType;
+import com.oceanbase.odc.service.sqlcheck.rule.MySQLAffectedRows;
+
+import lombok.NonNull;
+
+public class SQLAffectedRowsFactory implements SqlCheckRuleFactory {
+
+    private final JdbcOperations jdbc;
+
+    public SQLAffectedRowsFactory(JdbcOperations jdbc) {
+        this.jdbc = jdbc;
+    }
+
+    @Override
+    public SqlCheckRuleType getSupportsType() {
+        return SqlCheckRuleType.RESTRICT_SQL_AFFECTED_ROWS;
+    }
+
+    @Override
+    public SqlCheckRule generate(@NonNull DialectType dialectType, Map<String, Object> parameters) {
+        String key = getParameterNameKey("allowed-max-sql-affected-count");
+        if (parameters == null || parameters.isEmpty() || parameters.get(key) == null) {
+            return new MySQLAffectedRows(1000, jdbc);
```

Comment:
Why 1000?

Reply:
When there is no input value for this rule, the most conservative tactics is used by default, which is the default value of the production environment.

Reply:
OK, throw an exception instead.